crates/primitives/starknet/src/crypto/merkle_patricia_tree/merkle_tree.rs

```diff
@@ -99,6 +101,40 @@ pub enum ProofNode {
     Edge(EdgeProofNode),
 }
 
+/// Utility struct used for serializing.
+#[cfg(feature = "std")]
+#[derive(Debug, Serialize)]
+struct PathWrapper {
+    value: starknet_ff::FieldElement,
+    len: usize,
+}
+
+#[cfg(feature = "std")]
+impl Serialize for ProofNode {
```

Comment:
When is it serialized? And why is it never deserialized?

Reply:
Needed in the rpc, if you remove it you get the following :

```
#[rpc(server, namespace = "starknet")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Serialize` is not implemented for `RpcGetProofOutput`